Effective  grant management for users

diveshjain-phy opened the pull request.

JBorrow commented on soauth/database/group.py:
```diff
@@ -41,6 +41,38 @@ class Group(SQLModel, table=True):
+    grants: str = Field(default="")
+
+
+    def has_grant(self, grant: str) -> bool:
```
Please apply the changes from `main` that make sure the grants are lowercase (see changes in user)

JBorrow commented on soauth/database/user.py:
```diff
@@ -87,13 +87,33 @@ def remove_grant(self, grant: str):
+            if group.grants:
+                all_grants.update(group.grants.split())
+        
+        return set(all_grants)
```
You don't need this cast to set, it's already a set

JBorrow commented on soauth/service/groups.py:
```diff
@@ -43,6 +44,8 @@ async def create(
     members: list[UUID]
         The list of users who should initially be in the group.
+    grants: str
+        
```
Doc?

JBorrow commented:
Sorry - I formatted your code by accident. But I fixed the issue in your branch. I added the `include_groups` parameter to the `get_effective_grants` method that you have. In the case where we're just asking for group info, that's set to false.

diveshjain-phy commented:
> Sorry - I formatted your code by accident. But I fixed the issue in your branch. I added the `include_groups` parameter to the `get_effective_grants` method that you have. In the case where we're just asking for group info, that's set to false.

Thanks! The `include_groups` parameter is super clean way of solving this.

diveshjain-phy commented:
@JBorrow Didn't realize there were no tests set up for api. I have made changes and added endpoints. This might take a little bit more time but can use existing tests for soauth and api test for hippo to setup tests for api?

JBorrow commented:
It's okay to not have API tests as long as the functionality is in the service layer with high test coverage.

diveshjain-phy commented:
I have pushed the updates, how should i move next? The web layer?

JBorrow commented:
I can take over from here! If you can, please focus on reviewing the hippo code.
